File: src/main/java/edu/cit/estillore/MentorMatch/MentorMatch/tutorprofile/TutorProfileRequest.java
package edu.cit.estillore.MentorMatch.MentorMatch.tutorprofile;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.List;

/** Backing object for a MENTOR creating/editing their own profile (FR-002). */
@Data
public class TutorProfileRequest {

    @Size(max = 1000)
    private String bio;

    @NotEmpty(message = "At least one subject is required")
    @Valid
    private List<TutorSubjectRequest> subjects;
}

File: android/app/src/main/java/edu/cit/estillore/mentormatch/data/model/TutorProfileRequest.kt
package edu.cit.estillore.mentormatch.data.model

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class TutorProfileRequest(
    val bio: String? = null,
    val subjects: List<SubjectProficiency> = emptyList()
)
